unit/java_bytecode/java_bytecode_parse_generics/WildcardGenericFunctions.java

```diff
@@ -0,0 +1,54 @@
+public class WildcardGenericFunctions
+{
+  // Test a wild card generic type
+  public static void processSimpleGeneric(SimpleGeneric<?> x) {
+    assert(x.t.equals(null));
+  }
+
+  // Test a wildcard generic bound by an interface
+  public static void processUpperBoundInterfaceGeneric(SimpleGeneric<? extends BasicInterface> x) {
+    assert(x.t.getX() == 4);
+  }
+
+  // Test a wild card generic bound by a class
+  public static void processUpperBoundClassGeneric(SimpleGeneric<? extends Foo> x) {
+    assert(x.t.getX() == 4);
+  }
+
+  // It isn't legal to have an wild card with two upper bounds
+  // Per language spec on intersection types
+
+  public static void processLowerBoundGeneric(SimpleGeneric<? super Foo> x, Foo assign) {
+    x.t = assign;
+  }
+
+  // It is not legal Java to specify both an upper and lower bound
+  // public static void processBoundSuperClassGeneric(SimpleGeneric<? extends Object super Foo> x, Foo assign) {
+  //   x.t = assign;
+  // }
+
+  // Test a wild card generic bound by a class
+  // public static void processBoundClassGenericDoubleBound(SimpleGeneric<? extends Foo & BasicInterface> x) {
```

Comment:
if wildcards can have multiplle bounds, we should have this as test

Reply:
Sorry comment not clear - wildcards can not have multiple bounds - this java code doesn't compile :)